libs/grunt-loader.js

```diff
@@ -0,0 +1,31 @@
+var fs = require('fs');
+var path = require('path');
+
+module.exports = {
+	/**
+	 * Given a folder, attempts to load any grunt modules included in the child directory node_modules. Due to flattening
+	 * in NPM 3+, grunt-springroll's new modules could actually be installed in the game's node_modules folder rather than
+	 * the node_modules folder underneath grunt-springroll. load-grunt-tasks attempts to connect the package.json with the
+	 * sibling folder node_modules, which is why tasks have failed to load NPM 3 and beyond. This however avoids that and
+	 * simply matches filenames based on convention
+	 *
+	 * @param {grunt} grunt The grunt instance with which to load tasks
+	 * @param {string} folder The folder to load tasks from (will look for a node_modules as a child directory)
+	 * @return void
+	 */
+	load: function(grunt, folder) {
+    var lastDirectory = process.cwd();
+
+    process.chdir(folder);
```

Comment:
What does `process.chdir()` bring here? Isn't `folder` already an absolute path?

Reply:
If you look, there's actually two `process.chdir` calls, which work like `pushd` and `popd` in a shell. Based on what I understand, `grunt.loadNpmTasks` loads tasks only in the `node_modules` directly below the current working directory, which means that nested ones like `node_modules/grunt-springroll/node_modules/grunt-bower-task` will not get picked up (hence the two calls to this method - one from with the project root and one from the plugin root).

With NPM 2, this wasn't an issue because all of `grunt-springroll`'s dependencies were loaded into `node_modules/grunt-springroll/node_modules`. However, with NPM 3's flattening, this is not the case and modules will be instead be installed to just `node_modules`, with conflicting modules version being resolved via `node_modules/grunt-springroll/node_modules`.